scripts/jira_oauth_setup_script.py

```diff
@@ -0,0 +1,216 @@
+
+
+  How to use:
+
+  $ python3 jira_oauth_setup_script.py -h
+  $ python3 jira_oauth_setup_script.py PROJECT_ID JIRA_URL
+  $ python3 jira_oauth_setup_script.py -m PROJECT_ID JIRA_URL
+"""
+
+
+import argparse
+from requests_oauthlib import OAuth1Session
+from oauthlib.oauth1 import SIGNATURE_RSA
+from Crypto.PublicKey import RSA
+from google.cloud import secretmanager
+from google.api_core.exceptions import AlreadyExists
+
+
+def create_secret(client, project_id, secret_id):
+    """Create a new secret with the given name in Secret Manager. A secret
+    is a logical wrapper around a collection of secret versions. Secret
+    versions hold the actual secret material.
+
+    Args:
+        client: A Secret Manager client to use to create the secret
+        project_id: The id of the Google Cloud project in which to
+                    create the secret
+        secret_id: The name of the secret to create
+    """
+
+    parent = client.project_path(project_id)
+    response = client.create_secret(parent, secret_id, {
+        'replication': {
+            'automatic': {},
+        },
+    })
+    print('Created secret: {}'.format(response.name))
+
+
+def add_secret_version(client, project_id, secret_id, payload):
+    """
+    Add a new secret version to the given secret with the provided payload.
+
+    Args:
+        client: A Secret Manager client to use to add the secret version
+        project_id: The id of the Google Cloud project in which to
+                    add the secret version
+        secret_id: The name of the secret to add a new version to
+        payload: The payload of the new secret version
+    """
+
+    parent = client.secret_path(project_id, secret_id)
+
+    if not isinstance(payload, bytes):
+        payload = payload.encode('UTF-8')
+
+    response = client.add_secret_version(parent, {'data': payload})
+    print('Added secret version: {}'.format(response.name))
+
+
+def main():
+    # Parse command line arguments
+    parser = argparse.ArgumentParser(
+        description=('Generate necessary keys/secrets and set up Jira server and Secret '
+                     'Manager such that the Cloud Monitoring Jira integration app can '
+                     'authenticate a Jira client using OAuth.'))
+
+    parser.add_argument('project_id',
+                        help=('id of the Google Cloud project whose Secret Manager '
+                              'to store Jira OAuth secrets in.'))
+
+    parser.add_argument('jira_url',
+                        help=('URL of the Jira Server to setup OAuth for, e.g '
+                              'https://jira.atlassian.com'))
+
+    parser.add_argument('-m',
+                        action='store_true',
+                        help=('Use already generated private/public RSA keys called '
+                              'private.pem and public.pem'))
+
+    args = parser.parse_args()
+
+
+    # Create or load in RSA public and private keys
+    if args.m:
```

Comment:
Discussed offline: Decided too keep the option to load in so that in the case that the script crashes halfway, the user can load the previously generated keys in and not have to repeat the first step of creating the application link.